tests/test_arrow_dataset.py

```diff
@@ -0,0 +1,114 @@
+import os
+import tempfile
+from unittest import TestCase
+
+import numpy as np
+import pyarrow as pa
+
+from nlp.arrow_reader import BaseReader
+from nlp.info import DatasetInfo
+from nlp.splits import SplitDict, SplitInfo
+
+
+class ReaderTester(BaseReader):
+    """
+    Build a Dataset object out of Instruction instance(s).
+    This reader is made for testing. It mocks file reads.
+    """
+
+    def _get_dataset_from_filename(self, filename_skip_take):
+        """Returns a Dataset instance from given (filename, skip, take)."""
+        filename, skip, take = (
+            filename_skip_take["filename"],
+            filename_skip_take["skip"] if "skip" in filename_skip_take else None,
+            filename_skip_take["take"] if "take" in filename_skip_take else None,
+        )
+        pa_table = pa.Table.from_pydict({"filename": [filename + "_" + str(x) for x in np.arange(100).tolist()]})
+        if skip is not None and take is not None:
+            pa_table = pa_table.slice(skip, take)
+        return pa_table
+
+
+class BaseDatasetTest(TestCase):
```

Comment:
@thomwolf @lhoestq  - added 3 tests for `.map()` for now and 1 test for `.filter()`